### ISSUE TITLE
Unnecessary error in if overlap checking

### ISSUE THREAD

**Issue by JosephAbbey** [microsoft/TypeScript#46068]
# Bug Report

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about overlap checking

### 💻 Code

```ts
if (this.current_char == '\n') {
  this.pos.line++;
  this.pos.column = 1;
  this.advance();
  if (this.current_char == '\r') {
    this.advance();
  }
  return new Token(Tokens.TERMINATOR, '\n', this.pos);
}
```
This is the first in a long line of "if else"s inside a lexer function.

The variable "this.current_char" is updated in "this.advance".

### 🙁 Actual behavior

VScode and the playground put an error on the nested if statement saying "This condition will always return 'false' since the types '"\n"' and '"\r"' have no overlap.ts(2367)", even-though "this.current_char" changed in "this.advance".

### 🙂 Expected behavior

I would expect it to show no errors.

**Comment by MartinJohns** [comment]
Duplicate of #9998.
